core/src/BitMatrix.cpp

```diff
@@ -96,15 +97,7 @@ BitMatrix::setRegion(int left, int top, int width, int height)
 void
 BitMatrix::rotate180()
 {
-	BitArray topRow(_width), bottomRow(_width);
-	for (int i = 0; i < (_height + 1) / 2; i++) {
-		getRow(i, topRow);
-		getRow(_height - 1 - i, bottomRow);
-		topRow.reverse();
-		bottomRow.reverse();
-		setRow(i, bottomRow);
-		setRow(_height - 1 - i, topRow);
-	}
+	BitHacks::Reverse(_bits, _rowSize * 32 - _width);
```

Comment:
It won't yield the same result if _rowSize * 32 != _width, will it? Since there is padding in each row, thus the shifting must be done for each of them.